AcloConcurrency/AcloConcurrency/Collection/Views/CollectionView.swift

```diff
@@ -0,0 +1,65 @@
+import SwiftUI
+
+struct CollectionView: View {
+
+    @ObservedObject private var collectionModel: CollectionViewModel
+    @State private var activeNavigation = false
+    @State private var ingredientName: String = ""
+
+    init(collectionModel: CollectionViewModel){
+        self.collectionModel = collectionModel
+    }
+
+    var body: some View {
+        NavigationView{
+            ScrollView(.vertical, showsIndicators: false) {
+                HStack(spacing: 18){
+                    LazyVGrid(columns: [GridItem(.flexible())], spacing: 18) {
+                        ForEach(collectionModel.mockArray.indices){ index in
+                            CollectionCellView(
+                                numberInRow: index,
+                                imageURL: collectionModel.mockArray[index].ingredientImage,
+                                ingredientName: collectionModel.mockArray[index].ingredientName
+                            )
+                            .onTapGesture {
+                               ingredientName = collectionModel.mockArray[index].ingredientName
+                                activeNavigation  = true
+                            }
+                        }
+                    }
+                    .frame(maxWidth: .infinity, alignment: .top)
+                    LazyVGrid(columns: [GridItem(.flexible())], spacing: 18) {
+                        ForEach(collectionModel.mockArray.indices){ index in
+                            CollectionCellView(
+                                numberInRow: index + 1,
+                                imageURL: collectionModel.mockArray[index].ingredientImage,
+                                ingredientName: collectionModel.mockArray[index].ingredientName
+                            )
+                            .onTapGesture {
+                                ingredientName = collectionModel.mockArray[index].ingredientName
+                                activeNavigation  = true
+                            }
+                        }
+                    }
+                    .frame(maxWidth: .infinity, alignment: .top)
+                }
+                .padding(.horizontal, 16)
+                NavigationLink(isActive: $activeNavigation) {
+                    CocktailsView(ingredientName: ingredientName)
+                } label: {
+                    EmptyView()
+                        .frame(height: 1)
+                }
```

Comment:
Use ZStack. Frame(height: 1) - bad practice

Reply:
And you user label: EmptyView() a lot of times. Create public extension in UI folder